### PR TITLE
Fixed weather for inherited CfgWorlds

### PR DESCRIPTION
**Pull request by BaerMitUmlaut**
**When merged this pull request will:**
- Fix weather values for maps that inherit their CfgWorlds class from another map
- Restructure and simplify the getMapData function a little bit

### CODE REVIEW

**Comment by esteldunedain** [comment]
lgtm

**Comment by jonpas** [comment]
We should also use `toLower` everywhere, casing can change from build to build of those maps (even though unlikely, but better to be safe than sorry).

**Comment by PabstMirror** [comment]
`if (toLower worldName in ["fata", "Abbottabad"]) exitWith {` - opps

File: addons/weather/functions/fnc_getMapData.sqf [hunk 1]
```diff
@@ -35,23 +39,42 @@ GVAR(WindDirectionProbabilities) = [
...
-// Check if the weather data is defined in the map config
-if (isArray (configFile >> "CfgWorlds" >> worldName >> "ACE_TempDay")) exitWith {
-    GVAR(TempDay)   = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_TempDay");
-    GVAR(TempNight) = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_TempNight");
-    GVAR(Humidity)  = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_Humidity");
+// Get all non inherited arrays to filter maps that inherit from Stratis/Altis/Tanoa
+private _nonInheritedArrays = configProperties [configFile >> "CfgWorlds" >> _worldName, "isArray _x", false];
+// And check if any custom non-inherited weather is defined through config and use that if so
+if ((configFile >> "CfgWorlds" >> _worldName >> "ACE_TempDay") in _nonInheritedArrays) exitWith {
+    if (isArray (configFile >> "CfgWorlds" >> _worldName >> "ACE_TempDay")) then {
```
Comment (BaerMitUmlaut) [inline addons/weather/functions/fnc_getMapData.sqf]: If one weather value is defined I think it's safe to assume all of them are. I don't think these checks are necessary.

Comment (jonpas) [inline addons/weather/functions/fnc_getMapData.sqf]: I think they are good, this function only runs on mission start anyways.

Comment (PabstMirror) [inline addons/weather/functions/fnc_getMapData.sqf]: I added them just to be safe.
Before there was 2 separate `isArray` checks for wind and temp.
It's possible that some map only defines one or the other which could give us empty arrays.

**Pull request merged**